### PR TITLE
VZ-10309 Integrate VPO with modules - core changes

### PR DESCRIPTION
**Pull request by pfmackin**
First pass at full lifecycle implementation of VPO module integration

1. This is still controlled by a feature flag (default off)
2. Install, Update, Uninstall, Upgrade tested with Coherence
3. Only Upgrade from Module to Module works. 
4. Need module-operator fix for non-module to module upgrade (change coming in follow-up PR)
5. Only Helm overrides is copied to Module - structured config is a follow up
6. Original Verrazzano controller runs concurrently with new Verrazzano controller
7. There is some synchronization done between controllers, enough for Coherence to work
8. Module handler shim updates both Verrazzano CR component status and Module status

### CODE REVIEW

File: platform-operator/constants/constants.go [hunk 1]
```diff
@@ -271,13 +274,19 @@ const ThanosInternalUserName = "verrazzano-thanos-internal"
 const VerrazzanoPlatformOperatorHelmName = "verrazzano-platform-operator"
 
 // VerrazzanoCRNameAnnotation is the annotation for the verrazzano CR name
-const VerrazzanoCRNameAnnotation = "verrazzano.io/vz-cr-name"
+const VerrazzanoCRNameAnnotation = "module.verrazzano.io/vz-cr-name"
 
 // VerrazzanoCRNamespaceAnnotation is the annotation for the verrazzano CR namespace
-const VerrazzanoCRNamespaceAnnotation = "verrazzano.io/vz-cr-namespace"
+const VerrazzanoCRNamespaceAnnotation = "module.verrazzano.io/vz-cr-namespace"
```
Comment (michael-cico) [inline platform-operator/constants/constants.go:280]: This stuff was all inroduced in the firts pass right?  These are just changes to new annotations?

File: platform-operator/controllers/verrazzano/component/clusterapi/clusterapi_component.go [hunk 2]
```diff
@@ -96,6 +96,11 @@ func (c clusterAPIComponent) ShouldInstallBeforeUpgrade() bool {
 	return false
 }
 
+// ShouldUseModule returns true if component is implemented using a Module
+func (c clusterAPIComponent) ShouldUseModule() bool {
```
Comment (michael-cico) [inline platform-operator/controllers/verrazzano/component/clusterapi/clusterapi_component.go:100]: Nit, maybe use `IsModule` or `UsesModuleInstall`.  But not a hill I'm going to die on.

File: platform-operator/controllers/verrazzano/reconcile/upgrade_component.go [hunk 3]
```diff
@@ -53,6 +53,14 @@ func (r *Reconciler) upgradeComponents(log vzlog.VerrazzanoLogger, cr *installv1
 	// Loop through all of the Verrazzano components and upgrade each one.
 	// Don't move to the next component until the current one has been succcessfully upgraded
 	for _, comp := range registry.GetComponents() {
+		if comp.ShouldUseModule() {
+			// Ignore if this component is being handled by a Module
+			continue
+		}
+		if !comp.IsEnabled(spiCtx.EffectiveCR()) {
```
Comment (michael-cico) [inline platform-operator/controllers/verrazzano/reconcile/upgrade_component.go:60]: Is this a bug in general?  Upgrading a component that isn't enabled?

Comment (pfmackin) [reply platform-operator/controllers/verrazzano/reconcile/upgrade_component.go:60]: we skip it if it is disabled.

File: platform-operator/constants/constants.go (hunk 1 above)
Comment (pfmackin) [reply platform-operator/constants/constants.go:280]: yes, we will need an API review when we revisit the module api

File: platform-operator/controllers/verrazzano/component/clusterapi/clusterapi_component.go (hunk 2 above)
Comment (pfmackin) [reply platform-operator/controllers/verrazzano/component/clusterapi/clusterapi_component.go:100]: it is more that install, it is lifecycle.   I can change this in the future if we decide on a better name.

File: platform-operator/controllers/verrazzano/reconcile/upgrade_component.go (hunk 3 above)
Comment (pfmackin) [reply platform-operator/controllers/verrazzano/reconcile/upgrade_component.go:60]: To answer your question, yes i think it was a latent bug.  We should never upgrade a component that is disable, unless you can think of a reason why.

**Review by michael-cico — APPROVED** [review]

**Pull request merged**